src/Logic/CoreRules.ml

```diff
@@ -746,14 +746,38 @@ let rctxt_ev i c ju =
       mk_Eq (inst_ctxt c e1) (inst_ctxt c e2)
     else tacerror "ctxt_ev: bad event, expected equality"
   in
-  let ev = mk_Land (L.rev_append l (b::r)) in
+  let ev = {se.se_ev with ev_expr = mk_Land (L.rev_append l (b::r))} in
   let wfs = wf_gdef NoCheckDivZero (se.se_gdef) in
-  wf_exp NoCheckDivZero wfs ev;
-  let new_ju = { se with se_ev = { se.se_ev with ev_expr = ev } } in
+  wf_ev NoCheckDivZero wfs ev;
+  let new_ju = { se with se_ev = ev } in
   Rctxt_ev(i,c), [ { ju with ju_se = new_ju } ]
```

Comment:
This change is incomplete because the context might contain quantified variables, it also has to be adapted when we make events expressions.